src/app/login-form/login-form.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { AngularFire } from 'angularfire2';
+import { Router } from '@angular/router';
+
+@Injectable()
+export class LoginFormService {
+	constructor(public af: AngularFire, public router: Router) {
+	}
+
+	loginGithub(): void {
```

Comment:
This whole thing should probably look like
```
loginGithub(): Observable<firebase.Auth(or whatever type)> {
  return this.af.auth
    .filter(auth => !!auth.github.uid)
```
Basically, it returns an Observable which the component will then subscribe to. When this Observable emits new events, the component will be prompted to do what it wants